Sources/SPTDataLoader/SPTDataLoaderFactory.m

```diff
@@ -193,8 +237,10 @@ - (void)requestResponseHandler:(id<SPTDataLoaderRequestResponseHandler>)requestR
         request.cachePolicy = NSURLRequestReturnCacheDataDontLoad;
     }
 
-    @synchronized(self.requestToRequestResponseHandler) {
-        [self.requestToRequestResponseHandler setObject:requestResponseHandler forKey:request];
+    SPTDataLoaderInterceptorChain * chain =
+    [[SPTDataLoaderInterceptorChain alloc] initWithInterceptors:self.interceptors];
+    @synchronized(self.requestToInterceptorChain) {
+        [self.requestToInterceptorChain setObject:chain forKey:request];
```

Comment:
It can as it is right now. I was thinking ahead by adding Cancellation at some point and I thought to cancel the single interceptor that is running for a given request (given multiple ones can be in a different stage of the interceptor chain). On the other hand all the interceptors can be cancelled at the same time regardless if they are in use and demand to the implementation to properly handle it.